src/main/java/org/springframework/data/jpa/repository/query/AbstractJpaQuery.java

```diff
@@ -239,11 +240,11 @@ private Query applyEntityGraphConfiguration(Query query, JpaQueryMethod method)
 		JpaEntityGraph entityGraph = method.getEntityGraph();
 
 		if (entityGraph != null) {
-			Map<String, Object> hints = Jpa21Utils.tryGetFetchGraphHints(em, method.getEntityGraph(),
+			List<QueryHintValue> hints = Jpa21Utils.tryGetFetchGraphHints(em, method.getEntityGraph(),
 					getQueryMethod().getEntityInformation().getJavaType());
 
-			for (Map.Entry<String, Object> hint : hints.entrySet()) {
-				query.setHint(hint.getKey(), hint.getValue());
+			for (QueryHintValue hint : hints) {
```

Comment:
Instead of iterating over  `List`/`Map`, let's provide `QueryHints.forEach(BiConsumer<String, Object>)` so we can pass in a method reference that is called for each tuple and potentially for the same key multiple times.